rviz_rendering/src/rviz_rendering/mesh_loader_helpers/stl_loader.cpp

```diff
@@ -52,29 +52,17 @@ bool STLLoader::load(uint8_t * buffer, const size_t num_bytes, const std::string
   // check for ascii since we can only load binary types with this class
   std::string buffer_str = std::string(reinterpret_cast<char *>(buffer), num_bytes);
 
-  if (buffer_str.substr(0, 5) == std::string("solid")) {
-    // file says that it is ascii, but why should we trust it?
-
-    // check for "endsolid" as well
-    if (buffer_str.find("endsolid", 5) != std::string::npos) {
-      RVIZ_RENDERING_LOG_ERROR_STREAM(
-        "The STL file '" << origin << "' is malformed. It "
-          "starts with the word 'solid' and also contains the "
-          "word 'endsolid', indicating that it's an ASCII STL "
-          "file, but rviz can only load binary STL files so it "
-          "will not be loaded. Please convert it to a "
-          "binary STL file.");
-      return false;
-    }
-
-    // chastise the user for malformed files
-    RVIZ_RENDERING_LOG_WARNING_STREAM(
-      "The STL file '" << origin << "' is malformed. It starts "
-        "with the word 'solid', indicating that it's an ASCII "
-        "STL file, but it does not contain the word 'endsolid' so "
-        "it is either a malformed ASCII STL file or it is actually "
-        "a binary STL file. Trying to interpret it as a binary "
-        "STL file instead.");
+  if (buffer_str.substr(0, 5) == std::string("solid") &&
+    buffer_str.find("endsolid", 5) != std::string::npos)
```

Comment:
I can't say I love this; it means that for every solidworks STL file that is binary, we have to spend the time searching through the whole thing just to determine that it is not ASCII and doesn't contain `endsolid`.  That could be slow, especially on large STL files.

Is there anything more efficient we can do here?  The specification for STL is at http://www.fabbers.com/tech/STL_Format , maybe there are some clues on how we can do this more efficiently.

Reply:
I haven't changed the search method in this PR, the efficiency is exatly the same. The only thing that has changed is whether STL files starting with a "solid" but without an "endsolid", gets through the check without a warning.

I agree with this being inefficient, but I believe it would be better to address this in another PR or an issue for the time being.

Reply:
That's fair, can you please open an issue about it?

Reply:
Done - https://github.com/ros2/rviz/issues/919